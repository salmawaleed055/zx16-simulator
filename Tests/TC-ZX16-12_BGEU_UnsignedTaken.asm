# TC-ZX16-12_BGEU_UnsignedTaken.s
# Test Objective: Validate BGEU (Branch Greater Than or Equal Unsigned) instruction - branch taken
# x1=-5 (0xFFFB), x2=10 (0x000A), branch to PC+4 (0xFFFB >= 0x000A unsigned is true)
li x1, -5  # x1 = 0xFFFB
li x2, 10  # x2 = 0x000A
bgeu x1, x2, +4 # Current PC + 4 = 0x0004 + 4 = 0x0008
nop        # Skipped
li x1, 1   # Target
ecall 0x3FF