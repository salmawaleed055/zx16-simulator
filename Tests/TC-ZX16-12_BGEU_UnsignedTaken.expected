0x0000: F679    li x1, 65531
0x0002: 1471    li x2, 10
0x0004: 24D2    bgeu x1, x2, 0x0008
0x0008: 0271    li x1, 1
0x000A: FFC7    ecall 0x3FF

--- Final State ---
x0: 0x0000
x1: 0x0001
x2: 0x000A
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x0000
a1: 0x0000
PC: 0x000C
---------------------
Simulation finished.